feature_engine/creation/binary_combination_feature_reference.py

```diff
@@ -0,0 +1,250 @@
+                    "Number of items in new_variables_names must be equal to number of "
+                    "items in Reference_variables * intems in variables to combine * binary operations."
+                    "In other words, the transformer needs as many new variable names"
+                    "as reference variables and binary operations to perform over the variables to "
+                    "combine."
+                )
+
+        self.reference_variables = reference_variables
+        self.variables_to_combine = variables_to_combine
+        self.new_variables_names = new_variables_names
+        self.binary_operations = binary_operations
+
+    def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
+        """
+        Performs dataframe checks.
+
+        Args:
+            X: pandas dataframe of shape = [n_samples, n_features]
+            The training input samples.
+            Can be the entire dataframe, not just the variables to transform.
+
+            y: It is not needed in this transformer.
+            Defaults to None. Alternatively takes Pandas Series
+
+        Returns:
+            self
+        """
+        # check input dataframe
+        X = _is_dataframe(X)
+
+        # check variables to combine are numerical
+        self.variables_to_combine = _find_or_check_numerical_variables(
+            X, self.variables_to_combine
+        )
+        
+       # check reference_variables are numerical
+        self.reference_variables = _find_or_check_numerical_variables(
+            X, self.reference_variables
+        )
+
+        # check if dataset contains na
+        _check_contains_na(X, self.reference_variables)
+        _check_contains_na(X, self.variables_to_combine)
+
+        # cannot divide by 0, as will result in error
+        if 'sub' in self.binary_operations:
+            if X[self.reference_variables].isin([0]).any().any():
+                raise ValueError(
+                    "Some of the refence variables contain 0 values. Check and "
+                    "remove those before using this transformer."
+            )
+
+        result = []
+        result_name = []
+        for operation in self.binary_operations:
+            for combine in self.variables_to_combine:
+                for reference in self.reference_variables:
+                    if operation == "Sub":
+                            result.append(X[combine].values - X[reference].values)  
+                            result_name.append(str(combine)+"_"+operation+"_"+str(reference))
+                    elif operation == "Div":
+                            result.append(X[combine].values / X[reference].values)
+                            result_name.append(str(combine)+"_"+operation+"_"+str(reference))
+                    else: 
+                        result.append("At least one of the entered binary_operations is not supported. ")
+                        result_name.append(str(combine)+"_"+operation+"_"+str(reference))
```

Comment:
I don't think here we need a for loop.

If we do:
`x[self.variables_to_combine].sub[reference]`

this will subtract the `reference` for all the variables to combine and return a dataframe of size [rows, variables_to_combine]

See here: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.div.html#pandas.DataFrame.div

I think we can also do:
`x[self.variables_to_combine] - X[reference]`

and this will subtract reference from all variables in 1 step.

See the docs, it is explained there.

Reply:
Also, I am noticing now that the transformation, that is the substraction and division, should be part of transform.

In fit, I think we only create the variable names:
https://github.com/solegalli/feature_engine/blob/master/feature_engine/creation/mathematical_combination.py

Reply:
Done!